src/Popover/Popover.tsx

```diff
@@ -0,0 +1,235 @@
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useDismiss,
+    useRole,
+    useInteractions,
+    useMergeRefs,
+    Placement,
+    useId,
+    useFocus,
+} from "@floating-ui/react";
+import { ButtonHTMLAttributes, Dispatch, HTMLProps, ReactNode, SetStateAction, cloneElement, createContext, forwardRef, isValidElement, useContext, useLayoutEffect, useMemo, useState } from "react";
+
+import Styles from "./Popover.module.scss";
+
+interface PopoverOptions {
+    initialOpen?: boolean;
+    placement?: Placement;
+    modal?: boolean;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+}
+
+export function usePopover({
```

Comment:
custom hooks should ideally be in a separate folder, ./src/hooks or something. 
this file has alot of things in it. please create separate files for the sake of readability.